core/utils.py

```diff
@@ -97,6 +102,76 @@ def get_recent_contributors(count: int = 4) -> List[Dict[str, Any]]:
         return []
 
 
+def fetch_packages_yaml(url=None):
+    """
+    Fetch package data from YAML source.
+    
+    Parameters
+    ----------
+    url : str, optional
+        URL to fetch YAML from. If None, uses the default pyOpenSci GitHub URL.
+        
+    Returns
+    -------
+    list of dict
+        List of package dictionaries.
+        
+    Raises
+    ------
+    ContributorDataError
+        If data cannot be fetched or parsed.
+    """
+    if url is None:
+        url = "https://raw.githubusercontent.com/pyOpenSci/pyopensci.github.io/main/_data/packages.yml"
+    
+    try:
+        with urlopen(url) as response:
+            yaml_content = response.read().decode('utf-8')
+            packages = yaml.load(yaml_content)
+            
+        if not isinstance(packages, list):
+            raise ContributorDataError("YAML data should be a list of packages")
+            
+        logger.info(f"Successfully fetched {len(packages)} packages from {url}")
+        return packages
+        
+    except URLError as e:
+        logger.error(f"Failed to fetch packages from {url}: {e}")
+        raise ContributorDataError(f"Network error: {e}")
+    except YAMLError as e:
+        logger.error(f"Failed to parse YAML: {e}")
+        raise ContributorDataError(f"YAML parsing error: {e}")
+    except Exception as e:
+        logger.error(f"Unexpected error fetching packages: {e}")
+        raise ContributorDataError(f"Unexpected error: {e}")
+
+
+def get_recent_packages(count=3):
+    """
+    Get the most recently accepted packages.
+    
+    Parameters
+    ----------
+    count : int, default 3
+        Number of recent packages to return.
+        
+    Returns
+    -------
+    list of dict
+        List of recent package dictionaries, sorted by date_accepted descending.
+    """
+    try:
```

Comment:
IS your intent to filter the packages here? To me it looks like his code and the method above grab the package list, but they don't sort it by date accepted?

Notice here in the data we have a data-accepted field. We want to use that to select the 3 packages that were most recently accepted. 
https://github.com/pyOpenSci/pyopensci.github.io/blob/main/_data/packages.yml#L28